components/pedigree/resources/src/main/resources/pedigree/view/datepicker.js

```diff
@@ -1,264 +1,401 @@
+  *
+  * @Usage
+  *         var element = new Element('text', {'name':'fuzzy-date-picker'});
+  *         var options = {inputFormat : "MY", decadesEnabled : true};
+  *         var fuzzyDatePicker = new FuzzyDatePicker(element, options);
+  *
+  * @Events
+  *         datepicker:date:changed -> Fired. Occures when the user changes the date. Causes an xwiki:date:changed event to be fired as well.
+  *         datepicker:date:updated -> Listened. The fuzzy date picker will update its displayed value from its text element on this event.
+  */
+  var FuzzyDatePicker = Class.create({
+    initialize: function(input, options) {
+      this.options = options || {};
+      this.inputFormat = options.inputFormat || "YMD";
+      this.alwaysEnabled = options.alwaysEnabled || this.inputFormat != "YMD";
+      this.decadesEnabled = options.decadesEnabled || false;
+
+      if (!input) {return};
+      this._input = input;
+      this._input.hide();
+
+      this.initStrings();
+
+      this.container = new Element('div', {'class' : 'fuzzy-date-picker'});
+      this.container.observe("datepicker:date:changed", this.onDateChange.bind(this))
+      this.container.observe("datepicker:date:updated", this.updateDateFromInput.bind(this));
+      this._input.wrap(this.container);
+      this.createInputs();
+      this.insertInputs();
     },
+    createInputs : function() {
+      var yearElement = new Element('input', {'type': 'text', 'class' : 'text', 'name': 'year', 'placeholder': this.strings.yearPlaceholder});
+      var monthElement = new Element('input', {'type': 'text', 'class' : 'text', 'name': 'month', 'placeholder': this.strings.monthPlaceholder});
+      var dayElement = new Element('input', {'type': 'text', 'class' : 'text', 'name': 'day', 'placeholder': this.strings.dayPlaceholder});
+      this.inputs = {};
+      this.inputs.y = new FuzzyYearInput(yearElement, this.decadesEnabled);
+      this.inputs.m = new FuzzyMonthInput(monthElement, this.alwaysEnabled, this.inputs.y);
+      this.inputs.d = new FuzzyDayInput(dayElement, this.alwaysEnabled, this.inputs.y, this.inputs.m);
 
-    getZeroPaddedValueRange : function (start, end) {
-      var values = [];
-      if (start <= end) {
-          for (var v = start; v <= end; ++v) {
-              values.push({'value': v, 'text' : ("0" + v).slice(-2)});
-          }
-      } else {
-          for (var v = end; v <= start; --v) {
-              values.push({'value': v, 'text' : ("0" + v).slice(-2)});
-          }
-      }
-      return values;
     },
+    insertInputs : function(){
+      this.inputFormat.toLocaleLowerCase().split("").forEach(function(identifier){
+        this.container.insert(this.inputs[identifier].getElement());
+      }.bind(this));
+    },
+    updateDateFromInput : function() {
+      var dateObject = JSON.parse(this._input.value);
+      if (dateObject["year"] != null) {
+        this.inputs.y.setValue(dateObject["year"]);
+      } else if (dateObject["decade"] != null) {
```

Comment:
This is using an outdated date format I think, we no longer have a `decade` field, instead we have a `range` field

A few things need to be updated (here and in a few other places - at least everywhere there is aword "decade") to support current JSON Fuzzy Date format we use in both pedigree and PT